security/pkg/server/ca/authenticate/kube_jwt.go

```diff
@@ -83,3 +106,57 @@ func (a *KubeJWTAuthenticator) Authenticate(ctx context.Context) (*Caller, error
 		Identities: []string{fmt.Sprintf(identityTemplate, a.trustDomain, callerNamespace, callerServiceAccount)},
 	}, nil
 }
+
+func (a *KubeJWTAuthenticator) validateRemoteK8sJwt(targetJWT string) ([]string, error) {
+	tokenReview := &v1.TokenReview{}
+	if a.jwtPolicy == jwt.JWTPolicyThirdPartyJWT {
+		tokenReview.APIVersion = "authentication.k8s.io/v1"
+		tokenReview.Kind = "TokenReview"
+		tokenReview.Spec = v1.TokenReviewSpec{
+			Token:     targetJWT,
+			Audiences: []string{"istio-ca"},
+		}
+	} else if a.jwtPolicy == jwt.JWTPolicyFirstPartyJWT {
+		tokenReview.APIVersion = "authentication.k8s.io/v1"
+		tokenReview.Kind = "TokenReview"
+		tokenReview.Spec = v1.TokenReviewSpec{
+			Token: targetJWT,
+		}
+	} else {
+		return nil, fmt.Errorf("invalid JWT policy: %v", a.jwtPolicy)
+	}
+	remoteClients := a.mc.GetRemoteKubeControllers()
+	for k, v := range remoteClients {
+		reviewRes, err := v.AuthenticationV1().TokenReviews().Create(context.TODO(), tokenReview, metav1.CreateOptions{})
+		if err != nil {
+			log.Warnf("failed to validate the JWT against cluster %q: %v", k, err)
+			continue
+		}
+		if reviewRes.Status.Error != "" {
+			log.Warnf("the service account authentication returns an error: %v", reviewRes.Status.Error)
+			continue
+		}
+		inServiceAccountGroup := false
+		for _, group := range reviewRes.Status.User.Groups {
+			if group == "system:serviceaccounts" {
+				inServiceAccountGroup = true
+				break
+			}
+		}
+		if !inServiceAccountGroup {
+			log.Warnf("the token is not a service account")
```

Comment:
would be helpful to log found the JWT token in which cluster, however the token is not associated with a service account.

Reply:
fixed